src/v/pandaproxy/schema_registry/types.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+#include <fmt/format.h>
+
+namespace pandaproxy::schema_registry {
+
+enum class schema_type { avro = 0, json, protobuf };
+
+inline constexpr std::string_view to_string_view(schema_type e) {
+    switch (e) {
+    case schema_type::avro:
+        return "AVRO";
+    case schema_type::json:
+        return "JSON";
+    case schema_type::protobuf:
+        return "PROTOBUF";
+    }
+}
+
+///\brief A subject is the name under which a schema is registered.
+///
+/// Typically it will be "<topic>-key" or "<topic>-value".
+using subject = named_type<ss::sstring, struct subject_tag>;
+static const subject invalid_subject{};
+
+///\brief The definition of the schema.
+///
+/// TODO(Ben): Make this cheap to copy
+using schema_definition = named_type<ss::sstring, struct schema_definition_tag>;
```

Comment:
`schema_definition_view` based on a string_view?

Reply:
> `schema_definition_view` based on a string_view?

It'll probably be based on iobuf.